### PR TITLE
feat: move mako logic to init method

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description
This covers the edxmako implementation that was proposed in this pr https://github.com/eduNEXT/eox-nelp/pull/72 and add some extra changes like removing old implementation and adding unit testing and basically this allows to use any eox-nelp template outside a specific view. 

This doesn't include the payment notification template

## Testing instructions
1. Go to `/eox-nelp/frontend/experience/feedback/courses/`
2. Go to `/eox-nelp/stats/tenant/`
3. Go to the home page, this implementation requires the last version of the nelp bragi theme

### CODE REVIEW

File: eox_nelp/stats/views.py [hunk 1]
```diff
@@ -47,4 +47,4 @@ def get_tenant_stats(request):
     context = {query_param: "true" for query_param in STATS_QUERY_PARAMS}
     context.update(request.GET.dict())
 
-    return render_to_response("tenant_stats.html", context)
+    return edxmako.shortcuts.render_to_response("tenant_stats.html", context, "main", request)
```
Comment (johanv26) [inline eox_nelp/stats/views.py:50]: Why is need the arg "main" to this render_to_response method,

Comment (johanv26) [reply eox_nelp/stats/views.py:50]: Ok I think this is related to that main dirs or something to do it with that way.
https://github.com/eduNEXT/eox-nelp/pull/98/files#diff-ec9dfd84694298d271197a0909b7b62e0301e9df3dce5c29d0cec0a8255ebd42L19
https://github.com/eduNEXT/eox-nelp/blob/13ed2ffd30202ead25fb67dd7f798b1c6c0f8085/eox_nelp/templates_config.py#L19

File: eox_nelp/tests/test_init_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        """ Test that method calls the expected methods during the pipeline execution.
+
+        Expected behavior:
+            - set_mako_templates_mock is called once.
+            - patch_user_gender_choices_mock is called once.
+        """
+        run_init_pipeline()
+
+        set_mako_templates_mock.assert_called_once()
+        patch_user_gender_choices_mock.assert_called_once()
```
Comment (johanv26) [inline eox_nelp/tests/test_init_pipeline.py:34]: Ok this add gender choices called test xD

**Review by johanv26 — APPROVED** [review]
I didnt understood why I need the version of bragi theme. 
I think this is only an eox-nelp change with its views, by the way, the refactor is working for my local env.
Peek 2023-09-18 12-14

**Comment by andrey-canon** [comment]
> I didnt understood why I need the version of bragi theme. I think this is only an eox-nelp change with its views, by the way, the refactor is working for my local env. Peek 2023-09-18 12-14 [ Peek 2023-09-18 12-14 ](https://user-images.githubusercontent.com/51926076/268723300-c00eecfd-25b7-4617-b256-819382bb3812.gif) [ ](https://user-images.githubusercontent.com/51926076/268723300-c00eecfd-25b7-4617-b256-819382bb3812.gif)

There are two parts where the stats and feedback views are called, their own page and the home page so I mentioned the last bragi version since this [change ](https://github.com/eduNEXT/ednx-saas-themes/pull/179) just works with the version of eox-nelp that includes the payment notification version

**Pull request merged**